Source/EasyNetQ/Internals/AsyncLock.cs

```diff
@@ -26,26 +25,17 @@ public AsyncLock()
         {
             semaphore = new SemaphoreSlim(1);
             releaser = new Releaser(semaphore);
-            releaserTask = Task.FromResult(releaser);
         }
 
         /// <summary>
         /// Acquires a lock
         /// </summary>
         /// <param name="cancellationToken">The cancellation token</param>
         /// <returns>Releaser, which should be disposed to release a lock</returns>
-        public Task<IDisposable> AcquireAsync(CancellationToken cancellationToken = default)
+        public async Task<IDisposable> AcquireAsync(CancellationToken cancellationToken = default)
         {
-            var acquireTask = semaphore.WaitAsync(cancellationToken);
-            return acquireTask.IsCompleted
-                ? releaserTask
-                : acquireTask.ContinueWith(
-                    (_, state) => (IDisposable)state,
-                    releaser,
-                    default,
-                    TaskContinuationOptions.ExecuteSynchronously,
-                    TaskScheduler.Default
-                );
+            await semaphore.WaitAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Without this change, in case of cancellation, the releaser will be returned instead of OperationCancelledException.

Reply:
And it breaks the mutex semantic.